### PR TITLE
Desktop: Fixed auto scroll issue in editor

### LINKED ISSUE #1546: Inconvenient auto scroll when editing note
### Operating system
- Windows

### Application
- Desktop

First thanx for this great software.

I use a lot of code in my notes. I noticed that when I use piece of code
```
of this kind
```
or when I insert images
and when my notes are longer than one screen, the editing screen regularly scrolls to the top (step by  step). But finally it scrolls in such a way that the line with the cursor moves above the top of the screen. So the line where the cursor is currently focused disappears above the top and will reappear when I start typing again. After a few more typing it will scroll again above the top and so forth.
In appearance parameter I did not select any Editor font family.

### LINKED ISSUE #2702: Viewer scroll when adding text under a mermaid graph
## Environment

Joplin version: 1.93
Platform: Linux
OS specifics: Ubuntu 18.04

## Steps to reproduce

1. Create a note with this content.
```
```mermaid
graph TD;
	a-->b;
``
e
e
e
e
e
e
e
e
e
e
e
e
e
e
e
e
e
e
e
e
e
e
```

2. Scroll a bit down, just enough to hide a part of the mermaid graph.
3. Add text under the mermaid code.

The bigger the mermaid graph, the bigger the scrolling is.

### PR DESCRIPTION
**Pull request by Liuhaai**
The scroll event in the viewer might be triggered by some text input in the editor. Then scroll event will adjust editor's position, which causes auto scroll issue. I fixed this issue by a workaround. The editor position will not be changed by scroll event in the viewer until there is no content changes in the viewer. That is, the user is scrolling the viewer.
Fixed #1546
Fixed #2702 probably
Because this is just a workaround, I will look into position synchronization in the viewer side later.

### CODE REVIEW

File: ElectronClient/gui/note-viewer/index.html [hunk 1]
```diff
@@ -310,23 +310,29 @@
...
-				return;
-			}
-
-			lastScrollEventTime = 0;
-
-			const percent = currentPercentScroll();
-			setPercentScroll(percent);
 			
-			ipcProxySendToHost('percentScroll', percent);
+			// The editor position should change parallelly when the user scrolls on the viewer.
```
Comment (tessus) [inline ElectronClient/gui/note-viewer/index.html]: please remove `parallelly`

Comment (Liuhaai) [reply ElectronClient/gui/note-viewer/index.html]: Description fix:)

Comment (tessus) [reply ElectronClient/gui/note-viewer/index.html]: thx

**Comment by laurent22** [comment]
I can't replicate the bug #1546 with latest master anymore. If it's still happening for you, please could you provide a sample note and steps to replicate the issue?

Bug #2702 still happens on master and this pull request does not fix it.

**Comment by Liuhaai** [comment]
> I can't replicate the bug #1546 with latest master anymore. If it's still happening for you, please could you provide a sample note and steps to replicate the issue?
> 
> Bug #2702 still happens on master and this pull request does not fix it.

OK, I will look into it

**Comment by laurent22** [comment]
I fixed several scroll related issue in the latest version, including this one I think so I'll close the pull request. Thanks anyway for looking into it @Liuhaai!

**Pull request closed without merging**